processing/src/main/java/org/apache/druid/query/aggregation/AggregatorFactory.java

```diff
@@ -365,6 +365,17 @@ public AggregatorFactory withName(String newName)
     throw new UOE("Cannot change output name for AggregatorFactory[%s].", this.getClass().getName());
   }
 
+  /**
+   * @return true if {@link #getCombiningFactory()} of this aggregator can be used to combine the non-finalized results
+   * of the other aggregator. This does not imply that the opposite is true, the combining factory of the other
+   * aggregator might not necessarily be able to combine the intermediary results of this aggregator.
+   */
+  // todo (clint): naming things is the hardest, revisit this
+  public boolean canCombiningFactoryCombine(AggregatorFactory toCombine)
```

Comment:
The documentation and naming of this method isn't I think representing what we really need out of it. Like, what does "can be used to combine" mean? Someone might thing it means something, like, "the types are compatible" or "the aggregation function is compatible". But that's not really it. It's also looking at the inputs to the two factories.

IMO the concept of "substitute" should be part of how we either name or javadoc the method. I mean that what we really want to know is, if we have an intermediate aggregation produced by `toCombine`, could we substitute that for the one that this aggregator would produce?